
# Korelasi Data
1. Mengelola data menggunakan pandas

In [9]:
# import library panda
import pandas as pd

#import datasheet
from google.colab import files
uploaded = files.upload()



Saving 2778.csv to 2778.csv


In [10]:
#membaca file datasehet csv
df=pd.read_csv("2778.csv")
df

Nilai UAN  Presensi  ...  Nilai rata-rata matakuliah wajib semester 1   IPK
0          14        14  ...                                        37.62  3.96
1          18         9  ...                                        45.54  3.45
2          18        14  ...                                        61.38  3.96
3          10         4  ...                                        23.76  2.75
4          21         1  ...                                        74.25  3.01
5          10        11  ...                                        58.41  3.25
6          14         5  ...                                        74.25  2.87
7          24        11  ...                                        75.24  3.76
8          23         5  ...                                        27.72  2.85
9          25         0  ...                                        53.46  2.65
10         18         2  ...                                        53.46  3.86
11         23         3  ...                                        86.13  3.30
12         13         9  ...                                        95.04  3.21
13         15         1  ...                                        28.71  3.10
14         17        10  ...                                        87.12  2.76
15         22         5  ...                                        58.41  2.86
16         25        13  ...                                        70.29  2.55
17         11         5  ...                                        56.43  3.15
18         25         4  ...                                        51.48  3.45

[19 rows x 6 columns]

In [3]:
# ganti nama kolom
df_new = df.rename(columns={'Nilai rata-rata matakuliah wajib semester 1':'Matkul Wajib'})

#korelasi = df[['Nilai UAN','Presensi', 'Nilai TPA', 'Nilai TOEFL', 'Nilai rata-rata matakuliah wajib semester 1', 'IPK']]

# mencari nilai korelasi dengan metode pearson product momment
cormat = df_new.corr(method='pearson')
round(cormat, 10)

Nilai UAN  Presensi  ...  Matkul Wajib       IPK
Nilai UAN      1.000000 -0.155471  ...      0.164025 -0.093285
Presensi      -0.155471  1.000000  ...      0.201685  0.354966
Nilai TPA     -0.278212 -0.103233  ...     -0.366889 -0.268584
Nilai TOEFL   -0.025349 -0.026387  ...      0.059926  0.081863
Matkul Wajib   0.164025  0.201685  ...      1.000000 -0.034317
IPK           -0.093285  0.354966  ...     -0.034317  1.000000

[6 rows x 6 columns]

### Variasi nilai koefisien korelasi:
* 0.81 - 1 	: hubungan sangat kuat
* 0.61 - 0.8 	: hubungan kuat
* 0.41 - 0.6	: hubungan sedang
* 0.21 - 0.4	: hubungan lemah
* 0.01 - 0.2	: hubungan sangat lemah

### Penjelasan Hasil Korelasi Bari Pertama
* Nilai Koefisien korelasi Nilai UAN dengan Presensi adalah -0.155471 (negatif). Berlawanan arah, maksudnya apabila Nilai UAN tinggi maka belum tentu presensi nya tinggi, kemungkinan presensinya rendah
* Nilai koefisien korelasi Nilai UAN dengan Nilai TPA adalah -0.278212 (negatif). Berlawanan arah, maksudnya apabila Nilai UAN tinggi maka belum tentu Nilai TPA nya tinggi, kemungkinan Nilai TPA nya rendah
* Nilai koefisien korelasi Nilai UAN dengan Nilai TOEFL adalah -0.025349 (negatif). Berlawanan arah, maksudnya apabila Nilai UAN tinggi maka belum tentu Nilai TOEFL nya tinggi, kemungkinan Nilai TOEFL nya rendah
* Nilai koefisien korelasi Nilai UAN dengan Matkul Wajib adalah 0.164025 (positif). Searah, maksudnya apabila Nilai UAN tinggi, kemungkinan Nilai Matkul wajib juga tinggi. Namun karena nilai koefisiennya < 0.4, hubungannya lemah
* Nilai koefisien korelasi Nilai UAN dengan IPK adalah -0.093285 (negatif). Berlawanan arah, maksudnya apabila Nilai UAN tinggi maka belum tentu IPK nya tinggi, kemungkinan IPK nya rendah



# Regresi Linear Berganda

1. Inisialisasi variable dependence dan independence

In [4]:
import numpy as np

X = df_new[["Nilai UAN", "Presensi", "Nilai TPA", "Nilai TOEFL", "Matkul Wajib"]]
y = df_new[["IPK"]]


2. Mencari nilai intercept dan koefisien

In [5]:
from sklearn import linear_model

regression = linear_model.LinearRegression()
regression.fit(X, y)

print("a: ", regression.intercept_)
print("b: ", regression.coef_)

a:  [3.97980157]
b:  [[-0.00808231  0.03430779 -0.00186477  0.00057231 -0.00490933]]


3. Mencari Hasil regresi Nilai UAN

In [7]:
a = regression.intercept_
b = regression.coef_

In [8]:
# rumus: Y = a + bX
Y = a + b[0]
print(Y)

[3.97171926 4.01410936 3.9779368  3.98037388 3.97489224]
